src/PrestaShopBundle/Resources/config/services/bundle/security.yml

```diff
@@ -11,6 +11,25 @@ services:
             - "@prestashop.adapter.legacy.context"
             - "@prestashop.static_cache.adapter"
 
+    prestashop.security.admin.user_provider:
```

Comment:
I'm not sure if it's linked or not, but we have a `prestashop.adapter.data_provider.employee` service which is used in several places So does the services still works correctly with this new security management
Or should we replace it? Or maybe (to avoid too many BC breaks) use it as a proxy to the new `prestashop.security.admin.user_provider` service?

Reply:
By the way, maybe you could add a `deprecated` parameter in the service declarations? For `prestashop.security.admin.provider` and `prestashop.adapter.security.admin` at least but maybe there are others

Reply:
>I'm not sure if it's linked or not, but we have a prestashop.adapter.data_provider.employee service which is used in several places So does the services still works correctly with this new security management
Or should we replace it? Or maybe (to avoid too many BC breaks) use it as a proxy to the new prestashop.security.admin.user_provider service?

`prestashop.adapter.data_provider.employee` provides employee from current context, while the new `prestashop.security.admin.user_provider` is used to provide an employee by username and it's more related to security context

>By the way, maybe you could add a deprecated parameter in the service declarations? 

:+1:

Reply:
I suggest keeping these improvements for the next PR

Reply:
accepted, let's get this big PR merged or else we'll never finish it 😄(dont worry, I keep track of all of these "todo later" items)